src/snowflake/cli/api/entities/common.py

```diff
@@ -39,3 +40,13 @@ def perform(self, action: EntityActions, action_ctx: ActionContext):
         Performs the requested action.
         """
         return getattr(self, action)(action_ctx)
+
+
+_SQL_EXECUTOR: SqlExecutor | None = None
```

Comment:
Because the SqlExecutor has a connection, the extra global is not desirable -- we would then need to refactor all of this code when we allow for scoped CLI context with ContextVar. Can you think of a way to approach this without an additional global, or alternatively piggyback on top of cli context?

Reply:
There is an optional `SnowflakeConnection` constructor param. If it's not provided, the connection from CliGlobalContext will be used (which is also a singleton). Not sure I understand why it would require a larger refactor, can you explain?

We could also reuse a single `SqlExecutor` instance and provide it a connection (that's what I started with). But it required passing an extra param to each `ApplicationPackageEntity` method, and since the instance is pretty much stateless it was simpler to use a global singleton.

Reply:
Discussed f2f - I'm removing the global variable, and `get_sql_executor()` will return a new `SqlExecutor` instance every time.